cpp/oneapi/dal/algo/decision_forest/backend/model_impl.hpp

```diff
@@ -39,6 +67,40 @@ class detail::v1::model_impl : public base {
         return interop_;
     }
 
+    void traverse_dfs_impl(std::int64_t tree_idx, visitor_t&& visitor) const {
+        auto daal_model =
+            static_cast<const typename daal_model_map<Task>::daal_model_interop_t*>(interop_)
+                ->get_model();
+        if constexpr (std::is_same_v<Task, task::classification>) {
+            dt::visitor_interop<dtree_task_t> vis(std::move(visitor), class_count);
+            daal_model->traverseDFS(dal::detail::integral_cast<std::size_t>(tree_idx), vis);
+        }
+        else if constexpr (std::is_same_v<Task, task::regression>) {
+            dt::visitor_interop<dtree_task_t> vis(std::move(visitor));
+            daal_model->traverseDFS(dal::detail::integral_cast<std::size_t>(tree_idx), vis);
+        }
+        else {
+            static_assert(is_valid_task_v<Task>, "Unknown task");
+        }
+    }
+
+    void traverse_bfs_impl(std::int64_t tree_idx, visitor_t&& visitor) const {
+        auto daal_model =
```

Comment:
The same comments as for `traverse_dfs_impl`

Reply:
fixed